judge/views/submission.py

```diff
@@ -531,6 +541,37 @@ def get(self, request, *args, **kwargs):
         return super(ForceContestMixin, self).get(request, *args, **kwargs)
 
 
+class UserAllContestSubmissions(ForceContestMixin, AllUserSubmissions):
+    def get_title(self):
+        if self.is_own:
+            return _('My submissions in %(contest)s') % {'contest': self.contest.name}
+        return _("%(user)s's submissions in %(contest)s") % {
+            'user': self.username,
+            'contest': self.contest.name,
+        }
+
+    def access_check(self, request):
+        super().access_check(request)
+        if not self.contest.users.filter(user_id=self.profile.id).exists():
+            raise Http404()
+        if not self.contest.can_see_full_scoreboard(self.request.user):
+            if not self.is_own:
```

Comment:
Though this doesn't matter too much (since it will only matter for when you view your own submissions), maybe we should switch the order of these `if` statements, as computing `self.is_own` is cheap compared to `can_see_full_scoreboard` (not to mention, `is_own` is probably cached at this point).